src/riscv/lib/src/jit.rs

```diff
@@ -361,6 +371,29 @@ mod tests {
                 .expect("JIT compilation should succeed.");
         }
 
+        /// Run a test scenario in three modes:
+        ///
+        /// A) With the `setup_registers` preset (as if in a `setup_hook`)
+        /// B) With the `setup_registers` set in `Compressed` instructions
+        /// C) With the `setup_registers` set in `Uncompressed` instructions
+        fn run_with_setup_regs(&mut self) {
+            self.run();
+            let mut add_compressed = self.instructions.clone();
+            let mut add_uncompressed = self.instructions.clone();
+            if let Some(ScenarioSteps::N(exp_steps)) = self.expected_steps {
+                self.expected_steps =
+                    Some(ScenarioSteps::N(exp_steps + self.setup_registers.len()));
+            }
+            for (reg, value) in self.setup_registers.drain(..) {
+                add_compressed.insert(0, I::new_li(reg, value as i64, Compressed));
+                add_uncompressed.insert(0, I::new_li(reg, value as i64, Uncompressed));
```

Comment:
it _could_ be worth adjusting the `instr_pc` downwards when pushing new instructions? (so that the start of the 'user' block is at the instruction pc they set

maybe could add a potential TODO for that?